sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/BridgeInternal.java

```diff
@@ -418,6 +413,10 @@ public static CosmosResponseDiagnostics createCosmosResponseDiagnostics() {
         return new CosmosResponseDiagnostics();
     }
 
+    public static void setTransportClientRequestTimelineOnDiagnostics(CosmosResponseDiagnostics cosmosResponseDiagnostics, RequestTimeline requestTimeline) {
+        cosmosResponseDiagnostics.clientSideRequestStatistics().setTransportClientRequestTimeline(requestTimeline);
```

Comment:
What's the client stats context/scope? 
In-case the request is retried will it override existing timeline?

Reply:
It will be the transportclient stats of successful response

Reply:
Handled the scenario where we create multiple sub requests for single  request in case of consistency check on other replica. Now transportRequestTimeline is moved from parent clientSideRequestStatistics to StoreResult